internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestAdminOperationWithPreviousVersion.java

```diff
@@ -0,0 +1,869 @@
+    parentControllers = multiRegionMultiClusterWrapper.getParentControllers();
+
+    LOGGER.info(
+        "parentControllers: {}",
+        parentControllers.stream().map(VeniceControllerWrapper::getControllerUrl).collect(Collectors.joining(", ")));
+
+    int i = 0;
+    for (VeniceMultiClusterWrapper multiClusterWrapper: childClusters) {
+      LOGGER.info(
+          "childCluster{} controllers: {}",
+          i++,
+          multiClusterWrapper.getControllers()
+              .values()
+              .stream()
+              .map(VeniceControllerWrapper::getControllerUrl)
+              .collect(Collectors.joining(", ")));
+    }
+
+    clusterName = CLUSTER_NAMES[0];
+    VeniceControllerWrapper parentController = parentControllers.get(0);
+    parentControllerClient = new ControllerClient(clusterName, parentController.getControllerUrl());
+
+    // Pinning the version to the previous version
+    AdminTopicMetadataResponse updateProtocolVersionResponse =
+        parentControllerClient.updateAdminOperationProtocolVersion(
+            clusterName,
+            (long) (AdminOperationSerializer.LATEST_SCHEMA_ID_FOR_ADMIN_OPERATION - 1));
+    assertFalse(updateProtocolVersionResponse.isError(), "Failed to update protocol version");
+
+    veniceAdmin = multiRegionMultiClusterWrapper.getParentControllers().get(0).getVeniceAdmin();
+    PubSubTopicRepository pubSubTopicRepository = veniceAdmin.getPubSubTopicRepository();
+    TopicManager topicManager = veniceAdmin.getTopicManager();
+    PubSubTopic adminTopic = pubSubTopicRepository.getTopic(AdminTopicUtils.getTopicNameFromClusterName(clusterName));
+    topicManager.createTopic(adminTopic, 1, 1, true);
+
+    ControllerClient dc0Client = ControllerClient.constructClusterControllerClient(
+        clusterName,
+        multiRegionMultiClusterWrapper.getChildRegions().get(0).getControllerConnectString());
+    ControllerClient dc1Client = ControllerClient.constructClusterControllerClient(
+        clusterName,
+        multiRegionMultiClusterWrapper.getChildRegions().get(1).getControllerConnectString());
+    childControllerClients.add(dc0Client);
+    childControllerClients.add(dc1Client);
+
+    multiClusterWrapper = multiRegionMultiClusterWrapper.getChildRegions().get(0);
+  }
+
+  @AfterClass(alwaysRun = true)
+  public void cleanUp() {
+    AdminTopicMetadataResponse updateProtocolVersionResponse =
+        parentControllerClient.updateAdminOperationProtocolVersion(
+            clusterName,
+            (long) (AdminOperationSerializer.LATEST_SCHEMA_ID_FOR_ADMIN_OPERATION));
+    assertFalse(updateProtocolVersionResponse.isError(), "Failed to update protocol version");
+    multiRegionMultiClusterWrapper.close();
+
+    System.out.println(operationTypeMap);
+    for (Map.Entry<String, Boolean> entry: operationTypeMap.entrySet()) {
+      assertTrue(entry.getValue(), "Operation type " + entry.getKey() + " was not tested");
```

Comment:
BTW, if new schema introduces a new union payload, and we set writer schema to be n-1, how's the new test going to look like?

Reply:
Two actions:
1. add integration test for that schema type
2. in that test, add `markAsTested(<schema name>);`, else, this check will keep failing

Reply:
I updated the error message with steps to resolve the error if it happens.